pkg/collector/check/stats/stats.go

```diff
@@ -258,6 +260,41 @@ func (cs *Stats) SetStateCancelling() {
 	cs.Cancelling = true
 }
 
+func (cs *Stats) Clone() *Stats {
+	cloned := &Stats{
+		CheckName:                cs.CheckName,
+		CheckVersion:             cs.CheckVersion,
+		CheckConfigSource:        cs.CheckConfigSource,
+		CheckID:                  cs.CheckID,
+		Interval:                 cs.Interval,
+		LongRunning:              cs.LongRunning,
+		Cancelling:               cs.Cancelling,
+		TotalRuns:                cs.TotalRuns,
+		TotalErrors:              cs.TotalErrors,
+		TotalWarnings:            cs.TotalWarnings,
+		MetricSamples:            cs.MetricSamples,
+		Events:                   cs.Events,
+		ServiceChecks:            cs.ServiceChecks,
+		HistogramBuckets:         cs.HistogramBuckets,
+		TotalMetricSamples:       cs.TotalMetricSamples,
+		TotalEvents:              cs.TotalEvents,
+		TotalServiceChecks:       cs.TotalServiceChecks,
+		TotalHistogramBuckets:    cs.TotalHistogramBuckets,
+		EventPlatformEvents:      maps.Clone(cs.EventPlatformEvents),
+		TotalEventPlatformEvents: maps.Clone(cs.TotalEventPlatformEvents),
+		AverageExecutionTime:     cs.AverageExecutionTime,
+		LastExecutionTime:        cs.LastExecutionTime,
+		LastSuccessDate:          cs.LastSuccessDate,
+		LastError:                cs.LastError,
+		LastDelay:                cs.LastDelay,
+		LastWarnings:             slices.Clone(cs.LastWarnings),
+		UpdateTimestamp:          cs.UpdateTimestamp,
+		telemetry:                cs.telemetry,
+	}
+	copy(cs.ExecutionTimes[:], cloned.ExecutionTimes[:])
```

Comment:
This arguments should be the other way around: https://pkg.go.dev/builtin#copy

Would it make sense to use deepcopy package (already a dependency) to handle this and other deep-copying cases as well?

Reply:
Oh nice catch !

> Would it make sense to use deepcopy package to handle this

Sure, that would be safer !